### PR TITLE
Feature/#116 mode view

### LINKED ISSUE #116: Refactor: ModeModal -> ModeWindow + AddPlayerPanel

### PR DESCRIPTION
**Pull request by DominikNowak**
closes #116

### CODE REVIEW

File: src/app/MenuView.ts [hunk 1]
```diff
@@ -1,32 +1,38 @@
-import { ModeModal } from './components/ModeModal';
+import { PlayerDTO } from '~src/Interfaces/PlayerDTOInterface';
```
Comment (vieraboschkova) [inline src/app/MenuView.ts:1]: Remove tildas from paths?

File: src/app/ModeView.ts [hunk 2]
```diff
@@ -1,30 +1,41 @@
...
   private playerInputsWrapper: HTMLElement;
   private backButton!: HTMLElement;
   private playButton!: HTMLElement;
-  private submitCallback: CallbackOneParam<PlayerDTO[]>;
+  private submitCallback: CallbackTwoParam<boolean, PlayerDTO[]>;
+  private backCallback: CallbackNoParam;
 
   /**
-   * @param submitCallback - will be called onSubmit with PlayerDTO[] data in the argument
+   * @param submitCallback - will be called onSubmit with isDynamic and PlayerDTO[] data in the arguments
```
Comment (vieraboschkova) [inline src/app/ModeView.ts:22]: Good you remembered to change the definitions! 👍

File: src/app/ModeView.ts [hunk 3]
```diff
@@ -35,38 +46,65 @@ export class ModeModal extends BasicModal {
       'remove player',
     );
     this.playerInputsWrapper = Render.elementFactory('div', {
-      className: 'mode-inputs-wrapper',
+      className: 'mode-players-wrapper',
```
Comment (vieraboschkova) [inline src/app/ModeView.ts:49]: mode-players__wrapper?

File: src/app/ModeView.ts [hunk 4]
```diff
@@ -35,38 +46,65 @@ export class ModeModal extends BasicModal {
...
+      Render.elementFactory('input', {
+        className: 'mode-form__mode-input',
+        type: 'checkbox',
+        id: 'mode',
+        name: 'mode',
+      }),
+      Render.elementFactory(
+        'label',
+        {
+          className: 'mode-form__mode-input',
```
Comment (vieraboschkova) [inline src/app/ModeView.ts]: mode-form__label perhaps?

File: src/app/ModeView.ts [hunk 5]
```diff
@@ -35,38 +46,65 @@ export class ModeModal extends BasicModal {
...
+      'Click to add new player',
+    );
 
+    const mode = Render.elementFactory(
+      'div',
+      {
+        className: 'mode-form__mode',
+      },
+      Render.elementFactory('input', {
+        className: 'mode-form__mode-input',
```
Comment (vieraboschkova) [inline src/app/ModeView.ts:84]: I see you're nesting a lot, put perhaps just mode-form__input?

File: src/app/ModeView.ts [hunk 6]
```diff
@@ -35,38 +46,65 @@ export class ModeModal extends BasicModal {
...
+        },
+        'dynamic mode',
+      ),
+    );
     const form = Render.elementFactory(
       'form',
       {
         id: 'mode-form',
         action: '',
         method: 'get',
```
Comment (vieraboschkova) [inline src/app/ModeView.ts:103]: Is method necessary in this case?

**Review by vieraboschkova — DISMISSED** [review]
Good job!

**Review by vieraboschkova — APPROVED** [review]

**Review by Enessetere — APPROVED** [review]
Looks good. well done

**Pull request merged**